### PR TITLE
restructure documentation

### PR DESCRIPTION
**Pull request by evgeni**
see individual commit messages for their changes, TL;DR: move non-user docs to own files, remove "goals" section, move support sections further up

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,14 +2,6 @@
 
 Ansible modules for interacting with the Foreman API and various plugin APIs such as Katello.
 
-## Goals
```
Comment (ekohl) [inline README.md]: I still like the concept of goals. For example, it can clarify whether modules for plugin's resources are accepted or not.

Comment (evgeni) [reply README.md]: My original goal with dropping the goals section was to make the excerpt on Galaxy (https://galaxy.ansible.com/theforeman/foreman) more meaningful.

As to your explicit comment: doesn't the main description "Ansible modules for interacting with the Foreman API and various plugin APIs such as Katello" include all modules?

File: README.md [hunk 2]
```diff
@@ -2,25 +2,31 @@
...
-  * centralized location for community modules
-  * a single repository to clone for interacting with Foreman & plugins
-  * source for the official Ansible collection
-
 ## Documentation
 
 A list of all modules and their documentation can be found at [theforeman.org/plugins/foreman-ansible-modules](https://theforeman.org/plugins/foreman-ansible-modules/).
 
-## Supported Foreman and plugins versions
+Documentation how to [write](docs/developing.md), [test](docs/testing.md) and [debug](docs/debugging.md) modules is available in the [`docs`](docs/) folder.
```
Comment (mdellweg) [inline README.md]: Are those links working when viewed from Galaxy?
And i guess, i know the answer... Probably not.

Comment (evgeni) [reply README.md]: probably not, no.

Comment (evgeni) [reply README.md]: definitely not, just tested with the (old) `tests/README.md` link we had, and it points to https://galaxy.ansible.com/tests/README.md :see_no_evil:

**Comment by evgeni** [comment]
@ekohl @mdellweg any objections against merging after your/my comments?

File: README.md (hunk 1 above)
Comment (ekohl) [reply README.md]: I guess the summary on GH is also clear enough:
> Repository for Ansible modules that communicate with Foreman and plugin APIs

**Comment by mdellweg** [comment]
Apart from the link problem, I didn't look closely, but what I saw was reasonable.

**Comment by evgeni** [comment]
Given the link problem was there before, I am going to ignore it for now and see that we fix it in another way later.

**Comment by evgeni** [comment]
(I've opened #496 to track that)

**Pull request merged**